pyxem/signals/diffraction2d.py

```diff
@@ -478,80 +478,65 @@ def center_direct_beam(self,
                             *args, **kwargs)
 
     def remove_background(self, method,
-                          *args, **kwargs):
+                          **kwargs):
         """Perform background subtraction via multiple methods.
 
         Parameters
         ----------
         method : string
-            Specify the method used to determine the direct beam position.
-
-            * 'h-dome' -
-            * 'gaussian_difference' - Uses a difference between two gaussian
-                                convolutions to determine where the peaks are,
-                                and sets all other pixels to 0.
-            * 'median' - Use a median filter for background removal
-            * 'reference_pattern' - Subtract a user-defined reference patterns
-                from every diffraction pattern.
-
-        sigma_min : int, float
-            Standard deviation for the minimum gaussian convolution
-            (gaussian_difference only)
-        sigma_max : int, float
-            Standard deviation for the maximum gaussian convolution
-            (gaussian_difference only)
-        footprint : int
-            Size of the window that is convoluted with the array to determine
-            the median. Should be large enough that it is about 3x as big as the
-            size of the peaks (median only).
-        implementation : 'scipy' or 'skimage'
-            (median only) see expt_utils.subtract_background_median
-            for details, if not selected 'scipy' is used
-        bg : array
-            Background array extracted from vacuum. (subtract_reference only)
-        *args:
-            Arguments to be passed to map().
+            Specifies the method, from:
+            {'h-dome','gaussian_difference','median','reference_pattern'}
         **kwargs:
-            Keyword arguments to be passed to map().
```

Comment:
I basically quite like this structure, but do you think we should document the kwargs associated with each method (but not map) here?